### PR TITLE
QACTL Pytest classes

### PR DESCRIPTION
**Pull request by JoseAntonioMHerrera**
|Related issue|
|---|
|#1590|

## Description

Added the required classes to run successfully the group of tests from the configuration file:
- Pytest
- Test
- TestResult
- TestLauncher

Added the qa_run_tests.py that reads the configuration file and set all the necessary objects to run the required ansible tasks.

## Tests
- [X] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [X] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import os
+
+from wazuh_testing.provisioning.tests.TestResult import TestResult
+from wazuh_testing.provisioning.ansible.AnsibleRunner import AnsibleRunner
+from wazuh_testing.provisioning.ansible.AnsibleTask import AnsibleTask
+from wazuh_testing.provisioning.tests.Test import Test
+
+
+class Pytest(Test):
+    """ The class encapsulates the execution options of an especified set of tests and allows running them on the
```
[the remark this answers is not in the document]
Comment (antoniomanuelfr) [reply deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py:12]:
```suggestion
class Pytest(Test):
    """ The class encapsulates the execution options of a specified set of tests and allows running them on the
```

File: deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        if self.tests_result_path is None:
+            self.tests_result_path = os.path.join(tempfile.gettempdir(), '')
+        else:
+            self.tests_result_path = os.path.join(self.tests_result_path, '')
+
+        html_report_file_name = f"test_report-{datetime.now()}.html"
+        plain_report_file_name = f"plain_report-{datetime.now()}.txt"
+
+        shell = "python3 -m pytest "
```
Comment (antoniomanuelfr) [inline deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py:77]: This can be a class constant string. Check the previous comment

**Review by antoniomanuelfr — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py (hunk 2 above)
Comment (JoseAntonioMHerrera) [reply deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py]: Done in [commit](https://github.com/wazuh/wazuh-qa/pull/1675/commits/026d9a1b49c109624b462ed7678baab6f28aadc1)

File: deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoseAntonioMHerrera) [reply deps/wazuh_testing/wazuh_testing/provisioning/tests/Pytest.py]: Done in [commit](https://github.com/wazuh/wazuh-qa/pull/1675/commits/026d9a1b49c109624b462ed7678baab6f28aadc1)

File: deps/wazuh_testing/wazuh_testing/provisioning/qa_run_tests.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            except KeyError as e:
+                print("Error keyword: " + str(e))
+
+        return tests
+
+    def __init__(self, def_tests_file):
+        self.def_tests_file = def_tests_file
+
+
+run_pytest = RunQATests("tests/tests_def_file.yaml")
```
[the remark this answers is not in the document]
Comment (JoseAntonioMHerrera) [reply deps/wazuh_testing/wazuh_testing/provisioning/qa_run_tests.py]: Done in [commit](https://github.com/wazuh/wazuh-qa/pull/1675/commits/026d9a1b49c109624b462ed7678baab6f28aadc1)

File: deps/wazuh_testing/wazuh_testing/provisioning/qa_run_tests.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+            try:
+                if 'tests' in tests_obj:
+                    for key, value in tests_obj['tests'].items():
+                        tests.append(self.__build_test(value))
+                else:
+                    print("Malformed document. No tests root key founded.")
+
+            except KeyError as e:
+                print("Error keyword: " + str(e))
```
[the remark this answers is not in the document]
Comment (JoseAntonioMHerrera) [reply deps/wazuh_testing/wazuh_testing/provisioning/qa_run_tests.py]: Done in [commit](https://github.com/wazuh/wazuh-qa/pull/1675/commits/026d9a1b49c109624b462ed7678baab6f28aadc1)

File: deps/wazuh_testing/wazuh_testing/provisioning/qa_run_tests.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            except yaml.YAMLError as yaml_e:
+                print(f"Error while parsing: {yaml_e}", file=sys.stderr)
+                exit()
+
+            try:
+                if 'tests' in tests_obj:
+                    for key, value in tests_obj['tests'].items():
+                        tests.append(self.__build_test(value))
+                else:
+                    print("Malformed document. No tests root key founded.")
```
[the remark this answers is not in the document]
Comment (JoseAntonioMHerrera) [reply deps/wazuh_testing/wazuh_testing/provisioning/qa_run_tests.py]: Done in [commit](https://github.com/wazuh/wazuh-qa/pull/1675/commits/026d9a1b49c109624b462ed7678baab6f28aadc1)

**Review by jotacarma90 — APPROVED** [review]
LGTM

**Pull request merged**